### ISSUE TITLE
Doesn't support nested unique index.

### ISSUE THREAD

**Issue by ikus060** [bwaldvogel/mongo-java-server#9]
Creating a unique index with a sub-document using the dot notation doesn't work properly. No error are reported in the log.
```
dbCollection.createIndex(new BasicDBObject("action.actionid", 1), new BasicDBObject("unique", true));
```
Seams `UniqueIndex#getKeyValue(BSONObject)` is making a direct call to BSONObject#get() which doesn't support dot notation. I'm suggesting to make a call to Utils#getSubdocumentValue().

**Comment by bwaldvogel** [comment]
Thanks for reporting. Your description was pretty close to a pull request ;)

**Comment by bwaldvogel** [comment]
FYI: Just released version 1.3.0 which contains this fix.

**Comment by ikus060** [comment]
Your commit is causing more problem. When I query a document with the very same index, the backend doesn't find anything. It's obviously related to your change. I'm about to fetch your project into my eclipse to debug and fix it. I will soon find a fix.
